Source/HelixToolkit.Wpf.SharpDX/Model/Elements3D/LineGeometryModel3D.cs

```diff
@@ -426,34 +424,30 @@ public override void Dispose()
         /// </summary>
         private LinesVertex[] CreateLinesVertexArray()
         {
-            var positions = this.Geometry.Positions.Array;
-            var vertexCount = this.Geometry.Positions.Count;        
-            var color = this.Color;
+            var positions = Geometry.Positions.Array;
+            var vertexCount = Geometry.Positions.Count;        
+            var color = Color;
             var result = new LinesVertex[vertexCount];
+            var colors = Geometry.Colors;
 
-            if (this.Geometry.Colors != null && this.Geometry.Colors.Any())
+            for (var i = 0; i < vertexCount; i++)
             {
-                var colors = this.Geometry.Colors;
-
-                for (var i = 0; i < vertexCount; i++)
+                Color4 finalColor;
+                if (colors != null && colors.Any())
                 {
-                    result[i] = new LinesVertex
-                    {
-                        Position = new Vector4(positions[i], 1f),
-                        Color = color * colors[i],
-                    };
+                    finalColor = color*colors[i];
                 }
-            }
-            else
-            {
-                for (var i = 0; i < vertexCount; i++)
+                else
                 {
-                    result[i] = new LinesVertex
-                    {
-                        Position = new Vector4(positions[i], 1f),
-                        Color = color,
-                    };
+                    finalColor = color;
                 }
+
+                result[i] = new LinesVertex
+                {
+                    Position = new Vector4(positions[i], 1f),
+                    Color = finalColor,
+                    Parameters = new Vector4(IsSelected?1:0,0,0,0)

```

Comment:
`LinesVertex.Parameters` added to hold extra parameters for rendering. See the notes for `PointsVertex` below.